In [ ]:
import pandas as pd
import numpy as np

In [ ]:
cancer=pd.read_csv('/content/ovariantotal.csv')

In [ ]:
cancer.head()

,AFP,AG,Age,ALB,ALP,ALT,AST,BASO#,BASO%,BUN,...,PCT,PDW,PHOS,PLT,RBC,RDW,TBIL,TP,UA,TYPE
0,3.58,19.36,47,45.4,56,11,24,0.01,0.30,5.35,...,0.09,13.4,1.46,74,2.64,13.7,5.5,73.9,396.4,0
1,34.24,23.98,61,39.9,95,9,13,0.02,0.30,3.21,...,0.30,11.2,1.09,304,4.89,12.7,6.8,72.0,119.2,0
2,1.50,18.40,39,45.4,77,9,18,0.03,0.60,3.80,...,0.13,15.2,0.97,112,4.62,12.0,14.8,77.9,209.2,0
3,2.75,16.60,45,39.2,26,16,17,0.05,0.74,5.27,...,0.25,17.4,1.25,339,4.01,14.6,10.9,66.1,215.6,0
4,2.36,19.97,45,35.0,47,21,27,0.01,0.10,4.89,...,0.28,11.9,0.94,272,4.40,13.4,5.3,66.5,206.0,0


In [ ]:
cancer.columns

Index(['AFP', 'AG', 'Age', 'ALB', 'ALP', 'ALT', 'AST', 'BASO#', 'BASO%', 'BUN',
       'Ca', 'CA125', 'CA19-9', 'CA72-4', 'CEA', 'CL', 'CO2CP', 'CREA', 'DBIL',
       'EO#', 'EO%', 'GGT', 'GLO', 'GLU.', 'HCT', 'HE4', 'HGB', 'IBIL', 'K',
       'LYM#', 'LYM%', 'MCH', 'MCV', 'Menopause', 'Mg', 'MONO#', 'MONO%',
       'MPV', 'Na', 'NEU', 'PCT', 'PDW', 'PHOS', 'PLT', 'RBC', 'RDW', 'TBIL',
       'TP', 'UA', 'TYPE'],
      dtype='object')

In [ ]:
cancer.describe()

,AFP,AG,Age,ALB,ALP,ALT,AST,BASO#,BASO%,BUN,...,PCT,PDW,PHOS,PLT,RBC,RDW,TBIL,TP,UA,TYPE
count,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000,...,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000,349.000000
mean,11.388367,19.320516,45.051576,41.077077,77.083095,18.011461,19.103152,0.028883,0.479771,4.011117,...,0.251450,14.330487,1.120602,255.432665,4.359513,13.549914,9.089685,71.078797,243.706590,0.510029
std,95.090229,4.342422,15.130949,5.604478,43.733473,11.080940,8.372143,0.020890,0.346230,1.285569,...,0.085062,2.991921,0.187649,95.241974,0.465894,1.802818,4.104614,7.258465,68.581936,0.500617
min,0.610000,6.200000,15.000000,22.000000,26.000000,4.000000,7.000000,0.000000,0.000000,1.120000,...,0.070000,8.800000,0.570000,74.000000,2.620000,10.920000,2.500000,32.900000,96.000000,0.000000
25%,1.670000,17.300000,33.000000,38.500000,60.000000,11.000000,14.000000,0.010000,0.200000,3.160000,...,0.192000,11.900000,1.000000,202.000000,4.110000,12.400000,6.300000,67.500000,200.100000,0.000000
50%,2.420000,19.820000,45.000000,41.900000,71.000000,15.000000,18.000000,0.030000,0.400000,3.830000,...,0.240000,13.800000,1.120000,236.000000,4.370000,13.100000,8.600000,72.200000,235.400000,1.000000
75%,3.770000,22.160000,57.000000,45.100000,86.000000,20.000000,22.000000,0.040000,0.700000,4.780000,...,0.300000,16.800000,1.240000,292.000000,4.660000,14.400000,10.700000,75.800000,276.600000,1.000000
max,1210.000000,33.330000,83.000000,51.500000,763.000000,86.000000,78.000000,0.120000,1.940000,10.190000,...,0.690000,22.800000,1.750000,868.000000,6.740000,22.200000,38.300000,86.800000,632.000000,1.000000


In [ ]:
cancer.shape

(349, 50)

seperating features and labels using the drop function.

In [ ]:
x=cancer.drop('TYPE',axis=1)
y=cancer['TYPE']

In [ ]:
x.shape

(349, 49)

In [ ]:
y.shape

(349,)

In [ ]:
coll=cancer.columns
for i in coll:
  print(i)
  minc=cancer[i].min()
  maxc=cancer[i].max()
  print(minc,maxc)

AFP
0.61 1210.0
AG
6.2 33.33
Age
15 83
ALB
22.0 51.5
ALP
26 763
ALT
4 86
AST
7 78
BASO#
0.0 0.12
BASO%
0.0 1.94
BUN
1.12 10.19
Ca
0.92 2.83
CA125
3.75 5000.0
CA19-9
0.6 1000.0
CA72-4
0.2 158.5
CEA
0.2 138.8
CL
84.6 109.4
CO2CP
16.2 34.3
CREA
38.2 114.0
DBIL
0.9 12.1
EO#
0.0 0.4
EO%
0.0 7.6
GGT
4 176
GLO
14.1 47.6
GLU.
3.57 12.44
HCT
0.224 0.569
HE4
16.71 3537.6
HGB
61.8 189.0
IBIL
1.0 28.4
K
3.08 5.4
LYM#
0.35 3.49
LYM%
3.9 51.6
MCH
17.7 36.8
MCV
61.0 107.9
Menopause
0 1
Mg
0.65 1.37
MONO#
0.07 0.97
MONO%
0.3 21.3
MPV
5.06 14.5
Na
125.1 150.7
NEU
37.2 92.0
PCT
0.07 0.69
PDW
8.8 22.8
PHOS
0.57 1.75
PLT
74 868
RBC
2.62 6.74
RDW
10.92 22.2
TBIL
2.5 38.3
TP
32.9 86.8
UA
96.0 632.0
TYPE
0 1


In [ ]:
from sklearn import preprocessing
min_max_scaler=preprocessing.MinMaxScaler()
x_scale=min_max_scaler.fit_transform(x)
print(x_scale)

[[2.45578349e-03 4.85071876e-01 4.70588235e-01 ... 8.37988827e-02
  7.60667904e-01 5.60447761e-01]
 [2.78074070e-02 6.55363067e-01 6.76470588e-01 ... 1.20111732e-01
  7.25417440e-01 4.32835821e-02]
 [7.35908185e-04 4.49686694e-01 3.52941176e-01 ... 3.43575419e-01
  8.34879406e-01 2.11194030e-01]
 ...
 [1.83563615e-03 4.80280133e-01 6.47058824e-01 ... 3.15642458e-01
  6.58627087e-01 2.11753731e-01]
 [1.01704165e-03 7.74788058e-01 2.20588235e-01 ... 2.45810056e-01
  7.99628942e-01 3.53917910e-01]
 [8.26863129e-04 2.50645042e-01 3.52941176e-01 ... 1.62011173e-01
  6.62337662e-01 1.65858209e-01]]


Split the dataset:

In [ ]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test= train_test_split(x_scale,y,test_size=0.20,random_state=0)

In [ ]:
from tensorflow import keras
from keras.models import Sequential
from keras.layers import Dense

In [ ]:
model=Sequential()
model.add(Dense(32,activation='relu',input_dim=49))
model.add(Dense(32,activation='relu'))
model.add(Dense(1,activation='sigmoid'))

/usr/local/lib/python3.12/dist-packages/keras/src/layers/core/dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


Given:
Dataset size=349 samples


test_size=0.2 -> 20% for testing , so Training set=80% of 349 =279 samples(approx.)

batch_size=32

***steps per epoch= no.of samples/ batch size***(ceal of it)



In [ ]:
model.compile(optimizer='adam',loss='binary_crossentropy',metrics=['accuracy'])
model.fit(x_train,y_train,batch_size=32,epochs=15)

Epoch 1/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.4427 - loss: 0.7183
Epoch 2/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.6809 - loss: 0.6566 
Epoch 3/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.7143 - loss: 0.6328 
Epoch 4/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.7570 - loss: 0.6081 
Epoch 5/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.7409 - loss: 0.5920 
Epoch 6/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.7749 - loss: 0.5590 
Epoch 7/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.7838 - loss: 0.5237 
Epoch 8/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.7628 - loss: 0.5217 
Epoch 9/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.7220 - loss: 0.5354 
Epoch 10/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.7770 - loss: 0.4905 
Epoch 11/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.7952 - loss: 0.4750 
Epoch 12/15
9/9 ━━━━━━━━━━━━━━━━━━━━ 0s 9ms/step - accuracy: 0.8065 - loss: 0.4529 
Ep

In [ ]:
y_pred=model.predict(x_test)
y_pred

3/3 ━━━━━━━━━━━━━━━━━━━━ 0s 34ms/step


array([[0.6316353 ],
       [0.29358554],
       [0.6350377 ],
       [0.10894132],
       [0.7902053 ],
       [0.8881029 ],
       [0.8946245 ],
       [0.18544415],
       [0.0736198 ],
       [0.82953054],
       [0.75906825],
       [0.25768787],
       [0.8821554 ],
       [0.2761952 ],
       [0.07620823],
       [0.58563626],
       [0.6818639 ],
       [0.83017355],
       [0.92730427],
       [0.48652238],
       [0.7655341 ],
       [0.70724726],
       [0.2538118 ],
       [0.8708028 ],
       [0.0972137 ],
       [0.84483784],
       [0.03800626],
       [0.12960835],
       [0.05130719],
       [0.82862   ],
       [0.43709597],
       [0.38531253],
       [0.0627804 ],
       [0.11781386],
       [0.12026756],
       [0.01683688],
       [0.62678814],
       [0.8400136 ],
       [0.12842396],
       [0.23808986],
       [0.877794  ],
       [0.79254717],
       [0.78364426],
       [0.84113836],
       [0.7804726 ],
       [0.04434985],
       [0.2568346 ],
       [0.787

In [ ]:
y_pred=np.where(y_pred>0.5,1,0)
y_pred=y_pred.astype(int)
np.column_stack((y_pred,y_test))

array([[1, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [1, 1],
       [1, 1],
       [1, 1],
       [0, 0],
       [0, 0],
       [1, 1],
       [1, 1],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [1, 0],
       [1, 0],
       [1, 1],
       [1, 1],
       [0, 1],
       [1, 1],
       [1, 1],
       [0, 0],
       [1, 0],
       [0, 0],
       [1, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [0, 0],
       [1, 1],
       [1, 1],
       [0, 0],
       [0, 0],
       [1, 1],
       [1, 1],
       [1, 1],
       [1, 1],
       [1, 1],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [1, 0],
       [1, 1],
       [0, 1],
       [0, 0],
       [0, 0],
       [1, 1],
       [0, 0],
       [1, 1],
       [0, 0],
       [1, 0],
       [1, 1],
       [0, 1],
       [0, 0],
       [1, 1],
       [0, 0],
       [0, 0],
       [0,

In [ ]:
from sklearn.metrics import confusion_matrix,classification_report,accuracy_score
print(confusion_matrix(y_test,y_pred))
print(accuracy_score(y_test,y_pred))
print(classification_report(y_test,y_pred))

[[33  8]
 [ 3 26]]
0.8428571428571429
              precision    recall  f1-score   support

           0       0.92      0.80      0.86        41
           1       0.76      0.90      0.83        29

    accuracy                           0.84        70
   macro avg       0.84      0.85      0.84        70
weighted avg       0.85      0.84      0.84        70

